feat(dashboard): enable clickable sandboxes with detailed info modal

OxDTH opened the pull request:
## Description:

This PR introduces a new feature to the Sandbox dashboard that allows users to click on individual Sandbox rows to view detailed information in a modal. The implementation improves the user experience, enhances accessibility, and supports the support team's workflow with a copy-to-clipboard feature.

**Changes Made:**

1. `WorkspaceTable.tsx:`
 - Added `onRowClick` to support click handling.
 - Added logic to ignore clicks on interactive elements (buttons, checkboxes, links).

2. Added New File `SandboxInfoModal.tsx:`
- Displays detailed, categorized information such as Basic Info, Resources, Configuration and many more.
- Includes `Copy to Clipboard` for support team use.

- [ ] This change requires a documentation update
- [ ] I have made corresponding changes to the documentation

## Related Issue(s)

This PR addresses issue #1774 

## Screenshots/ScreenRecording:


https://github.com/user-attachments/assets/ec2dbff4-bb58-4dab-8af3-d3c5c462520d

## Notes

Users can click on any Sandbox row in the dashboard to view detailed information in a modal, and easily copy all the information to share with the support team.

fabjanvucina commented on apps/dashboard/src/components/SandboxInfoModal.tsx:
```diff
@@ -0,0 +1,240 @@
+      <DialogContent className="max-w-6xl max-h-[80vh] overflow-y-auto">
+        <DialogHeader>
+          <DialogTitle>Sandbox Information</DialogTitle>
+          <DialogDescription>Detailed information about sandbox {sandbox.id}</DialogDescription>
```
```suggestion
          <DialogDescription>Complete overview of the Sandbox details, resources and configuration</DialogDescription>
```

fabjanvucina commented on apps/dashboard/src/components/SandboxInfoModal.tsx:
```diff
@@ -0,0 +1,240 @@
+                <span className="text-muted-foreground">ID:</span>
+                <span className="font-mono break-all">{sandbox.id}</span>
+
+                <span className="text-muted-foreground">Name:</span>
```
We don't want to display the name

fabjanvucina commented on apps/dashboard/src/components/SandboxInfoModal.tsx:
```diff
@@ -0,0 +1,240 @@
+            ) : (
+              <>
+                <Copy className="w-4 h-4" />
+                Copy to Clipboard
```
```suggestion
                Copy ID
```

fabjanvucina commented on apps/dashboard/src/components/SandboxInfoModal.tsx:
```diff
@@ -0,0 +1,240 @@
+  sandbox: Workspace | null
+}
+
+export const SandboxInfoModal: React.FC<SandboxInfoModalProps> = ({ open, onOpenChange, sandbox }) => {
```
The component should be called `SandboxInfoDialog` instead of `..Modal`

fabjanvucina reviewed (changes requested):
@OxDTH Great work! I added a few comments in my review.

Also, could you please do the following:
- Rebase your branch because we added `autoArchiveInterval` as a new sandbox property
- Attach a screenshot of what the dialog would look like when there is an error reason and all values in the 'Configuration' category are populated.

OxDTH commented:
@fabjanvucina I have rebased and added screenshots of what the dialog would look like when there is an error.
Also, Fixed the requested changes.

![Screenshot 2025-06-05 at 3 07 32 PM](https://github.com/user-attachments/assets/6e2a47e6-bfab-4e1e-a729-4ea33095a6ea)

![Screenshot 2025-06-05 at 3 07 47 PM](https://github.com/user-attachments/assets/67fa00d0-1539-49c8-99d2-bd5d936c6b81)
